### PR TITLE
Update documentation for Dask cuDF

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
General documentation update for Dask cuDF:

- Adds `README.md` file to `dask_cudf` (this is currently a symlink to cudf's README, which isn't terribly helpful)
- Emphasizes direct usage of the `dask.dataframe` API (rather than the explicit `dask_cudf` API)
  - Including the `to_backend` API 
- Advertises query-planning support
- Includes a simple Dask CUDA example (and best-practices link)

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by rjzamora** [comment]
cc @quasiben @pentschev @wence- @charlesbluca @madsbk (happy to get feedback from any of you with time and interest)

**Review by pentschev — COMMENTED** [review]
Overall LGTM, thanks @rjzamora !

I've left some suggestions on a single comment, for whatever reason GH is not letting me comment on each line individually.

File: python/dask_cudf/README.md
[the remark this answers is not in the document]
Comment (rjzamora) [reply python/dask_cudf/README.md:1]: >For some reason GH is not allowing me to comment on each line individually, so I'm doing it all here
Weird!

>Line 116: result.compute()  # ... is it intentional that you're calling compute()? That is generally a suboptimal practice and having that in official docs tend to encourage that practice.

Good catch! That was supposed to be computing a groupby aggregation, but it seems I copied over the wrong example.

**Review by madsbk — APPROVED** [review]
LGTM

File: python/dask_cudf/README.md
[the remark this answers is not in the document]
Comment (rjzamora) [reply python/dask_cudf/README.md:1]: Okay, I changed the example to a groupby aggregation (which is typically safe to compute), and added this note:

```
> [!NOTE]
> This example uses `compute` to materialize a concrete `cudf.DataFrame` object in local memory. Never call `compute` on a large collection that cannot fit comfortably in the memory of a single GPU! See Dask's [documentation on managing computation](https://distributed.dask.org/en/stable/manage-computation.html) for more details.
```

[the remark this answers is not in the document]
Comment (bdice) [reply python/dask_cudf/README.md:1]: Same issue here, I can't comment on the diff lines. Replacing a symlink with a separate file is probably an edge case for the GitHub UI.

General comment: sometimes this says "The user" and sometimes it says "You." Let's use a consistent tense/audience.

Line 105: Just creating the `client` variable is enough? You don't have to pass it in or register it somewhere? I am not used to this pattern.

**Review by bdice — APPROVED** [review]
Thank you very much for writing this!

**Comment by bdice** [comment]
Crud. I didn't realize my approval would merge this PR. @rjzamora Can you apply my suggestions here? https://github.com/rapidsai/cudf/pull/16671#discussion_r1735013275

Also we need to make sure this information is all in sync with the dask-cudf docs, which don't mention query planning and other important topics. https://docs.rapids.ai/api/dask-cudf/stable/

**Comment by rjzamora** [comment]
>Oops. I didn't realize my approval would merge this PR

Yeah, my fault! No worries - I will submit a follow-up today to update the API docs and I'll include your suggestion to use consistent language.

File: python/dask_cudf/README.md
[the remark this answers is not in the document]
Comment (rjzamora) [reply python/dask_cudf/README.md:1]: >General comment: sometimes this says "The user" and sometimes it says "You." Let's use a consistent tense/audience.

Good point -  I'll address this in a follow up.

>Line 105: Just creating the client variable is enough? You don't have to pass it in or register it somewhere? I am not used to this pattern.

Yes, this is the common/suggested pattern in dask (unless you are working with multiple clusters at the same time. Which is extremely rare)

**Pull request merged**